subprojects/testing-jvm/src/main/java/org/gradle/api/internal/tasks/testing/junit/JUnitSpec.java

```diff
@@ -22,11 +22,13 @@
     private final Set<String> includeCategories;
     private final Set<String> excludeCategories;
     private final Set<String> includedTests;
+    private final String requestClass;
 
-    public JUnitSpec(Set<String> includeCategories, Set<String> excludeCategories, Set<String> includedTests) {
+    public JUnitSpec(Set<String> includeCategories, Set<String> excludeCategories, Set<String> includedTests, String requestClass) {
```

Comment:
@diacone It's a valid point you are making. However, we were putting more weight on the backward compatibility and removing the need to modify an extra file by passing empty string.